doc/io_streams.rdoc

```diff
@@ -326,23 +396,112 @@ that determine how lines in a stream are to be treated:
 
 - +:chomp+: If +true+, line separators are omitted; default is +false+.
 
-=== Open and Closed \IO Streams
+=== Character \IO
 
-A new \IO stream may be open for reading, open for writing, or both.
+You can process an \IO stream character-by-character using these methods:
 
-You can close a stream using these methods:
+- IO#getc: Reads and returns the next character from the stream:
 
-- IO#close: Closes the stream for both reading and writing.
-- IO#close_read (not available in \ARGF): Closes the stream for reading.
-- IO#close_write (not available in \ARGF): Closes the stream for writing.
+    f = File.new('t.rus')
+    f.getc # => "т"
+    f.getc # => "е"
+    f.getc # => "с"
+    f.getc # => "т"
+    f.getc # => nil
 
-You can query whether a stream is closed using these methods:
+- IO#readchar (not in \StringIO):
+  Like #getc, but raises an exception at end-of-stream:
 
-- IO#closed?: Returns whether the stream is closed.
+    f.readchar # Raises EOFError.
+
+- IO#ungetc (not in \ARGF):
+  Pushes back ("unshifts") a character or integer onto the stream:
+
+    f = File.new('t.rus')
+    c = f.getc         # => "т"
+    f.ungetc(c)
+    f.ungetc(c)
+    f.read             # => "ттест"
+    f.close
+    File.read('t.rus') # => "тест"
+
+- IO#putc (also in Kernel): Writes a character to the stream:
+
+    c = File.new('t.rus').getc # => "т"
+    f = File.new('t.tmp', 'w')
+    f.putc(c)
+    f.putc(c)
+    f.close
+    File.read('t.tmp')         # => "тт"
```

Comment:
It seems like the only purpose of `c` is to be a character? Why not just do this:

```suggestion
    f = File.new('t.tmp', 'w')
    f.putc("т")
    f.putc("т")
    f.close
    File.read('t.tmp')         # => "тт"
```

Reply:
Done.